pandas/core/dtypes/concat.py

```diff
@@ -89,10 +89,9 @@ def concat_compat(to_concat, axis=0):
     # filter empty arrays
     # 1-d dtypes always are included here
     def is_nonempty(x):
-        try:
-            return x.shape[axis] > 0
-        except Exception:
+        if x.ndim <= axis:
```

Comment:
Hmm I guess this replicates existing behavior but is this really right? Seems counter intuitive to return `True` in a function called `is_nonempty` if the axis doesn't exist at all

Reply:
the comment just above the function "1-d dtypes always are included here" seems to address that.  i agree it could be clearer, but I'm not going to lose sleep over it.  ideas for better wording?